source/utilities.cc

```diff
@@ -53,6 +54,222 @@ namespace aspect
+                  if (keyword_labels == "background" || keyword_labels == "bg")
+                    {
+                      // Overwrite the default x_mapped_values to user input.
+                      // Element_0 is reserved for the background.
+                      x_mapped_values[0] = (Utilities::string_to_double(split_parts[1]));
+                    }
+                  else
+                    {
+                      // Ensure that each non-special keyword found is also contained in
+                      // the Compositional fields subsection list, and insert the associated
+                      // values to the correct index position.
+                      for (unsigned int ifield=0; ifield!=list_of_field_names.size(); ++ifield)
+                        {
+                          if (keyword_labels.compare(list_of_field_names[ifield]) == 0)
+                            {
+                              // Throw an error if a match was already found...there can be only one
+                              AssertThrow (std::isnan(x_mapped_values[ifield+1]) == true,
+                                           ExcMessage ("The keyword <"
+                                                       + keyword_labels
+                                                       + "> in "
+                                                       + field_name
+                                                       + " is listed multiple times. "
+                                                       "Check that the field id list is "
+                                                       "the same as what is found in "
+                                                       "Compositional fields subsection "
+                                                       "from `set Names of fields'."
+                                                       "Alternatively, if `set Names of fields' "
+                                                       "is not set, the default names are "
+                                                       "C_1, C_2, ..., C_n."));
+
+                              // We passed...capture the value
+                              x_mapped_values[ifield+1] = (Utilities::string_to_double(split_parts[1]));
+                            }
+                        }
+                    }
+                }
+
+              // Go through one more time to make sure everything was set
+              for (unsigned int ifield=0; ifield!=n_fields_in; ++ifield)
+                {
+                  // Ensure that the special keyword "background" or "bg" does not
+                  // exist when there are the same number of fields as what
+                  // is set in the Compositional fields subsection.
+                  if (ifield == 0)
+                    {
+                      AssertThrow (std::isnan(x_mapped_values[ifield]) == false,
```

Comment:
You can move this out of the loop and just call `AssertThrow (std::isnan(x_mapped_values[0]) == false,` and then make the loop go from 1 to n_fields_in